src/a.tex

```diff
@@ -428,14 +428,28 @@ \section{Atomic Memory Operations}
 
 \vspace{-0.1in} The atomic memory operation (AMO) instructions perform
 read-modify-write operations for multiprocessor synchronization and
-are encoded with an R-type instruction format.  These AMO instructions
-atomically load a data value from the address in {\em rs1}, place the
-value into register {\em rd}, apply a binary operator to the loaded
-value and the original value in {\em rs2}, then store the result back
-to the original address in {\em rs1}. AMOs can either operate on 64-bit (RV64
-only) or 32-bit words in memory.  For RV64, 32-bit AMOs always
-sign-extend the value placed in {\em rd}, and ignore the upper 32 bits
-of the original value of {\em rs2}.
+are encoded with an R-type instruction format. A data value is atomically
+loaded from the address in {\em rs1} and placed into {\em rd}, then a
+binary operator is applied to the loaded value and the original value in
+{\em rs2}, finally the result is stored back to the original address.
+
+\begin{commentary}
+If {\em rs1} and {\em rd} refer to the same register, the address value
+in there is overwritten with the data value read from this address.
+However, this does not affect the address used for the final write
+operation, as AMOs always operate on a single memory address only.
```

Comment:
This paragraph can be reduced to just a few words: "original address in rs1".

Reply:
Well, this is the whole thing this PR is about. Unless you are are RISC-V expert, this is nothing really trivial. I think the docs should also help non-experts.

Reply:
The spec assumes you understand how CPUs generally work. In my review here, I'm trying to explain how the original wording was already complete and precise. If the spec had to spell out all these possibilities in such painstaking detail, it would be 4000 pages long. Brevity is important.

Reply:
I disagree here, the spec is also something used by average developers and students to just look up things. However, I can see your point. Seems there is a certain need to to versions then:
* a "pure" specs, that is technically unambiguous and ideally formalized. It can be used as input for tools then also.
* the verbose RISC-V Handbook" for humans. 

Note that I've added my comment in a commentary section on purpose, it's not pure specs but useful background information for human seeking help.

Reply:
This is a long-standing debate over the RISC-V ISA Manual. Some argue for more commentary, background, rationale, etc., and others want to cut all the fat. I acknowledge both sides. All I will say here is my observation that the community strongly resists to making changes in this document, especially ratified portions, without a clear argument that they address a defect. In this case, I don't see a convincing argument that there exists a defect.

I'd also like to respond to your respond to your "two versions" comment:

- In case you aren't aware, there is an ongoing effort, [RISC-V Sail Model](https://github.com/riscv/sail-riscv), to express the RISC-V in a formal language. (Some might also argue that the [Spike RISC-V ISA simulator](https://github.com/riscv-software-src/riscv-isa-sim) technically qualifies as a formal-language specification.)

- Patterson-Waterman also wrote a small book, [The RISC-V Reader](http://riscvbook.com/), which is intended to serve as a friendlier introduction. I believe they are in the process of producing a second edition.

It's tempting to propose an open, crowd-sourced "RISC-V Handbook", where detailed commentary like in this PR would certainly be welcome. But this repository is the wrong venue to pursue that.

Reply:
Thanks for the feedback. I'm closing this PR based on the comments. In case there is an option to contribute to [The RISC-V Reader](http://riscvbook.com) I'd be willing to contribute there.